packages/ra-ui-materialui/src/list/datagrid/ExpandAllButton.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { memo } from 'react';
+import IconButton from '@mui/material/IconButton';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { useTranslate, useExpandedMultiple } from 'ra-core';
+import { DatagridClasses } from './useDatagridStyles';
+import clsx from 'clsx';
+
+interface ExpandAllButtonProps {
+    resource: string;
+    ids: string[] | number[];
+}
+
+const ExpandAllButton = ({ resource, ids }: ExpandAllButtonProps) => {
+    const translate = useTranslate();
+    const [expanded, toggleExpanded] = useExpandedMultiple(resource, ids);
+
+    return (
+        <IconButton
+            className={clsx(DatagridClasses.expandIcon, {
+                [DatagridClasses.expanded]: expanded,
+            })}
+            aria-label={translate(
+                expanded ? 'ra.action.close' : 'ra.action.expand'
+            )}
+            aria-expanded={expanded}
+            tabIndex={-1}
```

Comment:
Why this tabindex?

Reply:
dito

Reply:
since ExpandRowButtons are not tab-selectable, the least i would do here is to stay consistent and keep tabindex = -1.